templates/server/src/main/resources/archetype-resources/core/src/main/resources/config/application.properties

```diff
@@ -27,6 +27,9 @@ spring.datasource.url=jdbc:mysql://address=(protocol=tcp)(host=localhost)(port=3
 #elseif ($dbType == 'mariadb')
 spring.datasource.password=todo
 spring.datasource.url=jdbc:mariadb://localhost:3306/db
+#elseif ($dbType == 'hana')
+spring.datasource.password=todo
+spring.datasource.url=jdbc:sap://localhost:39015
```

Comment:
As explained [here](https://www.sap.com/india/developer/tutorials/hxe-howto-tutorialprep.html), there is variation in port number as per the version of database. I would suggest not to hard code the port number and let the user figure out what port number to use. Otherwise chances are that users(especially junior members of the team) will be confused why he/she is getting connection error without realizing that the port number configured is not correct for their setup.

Reply:
39015 is the default port of the HANA express cloud and VM images. IMHO this is a sensible default. I'd just add a comment how to adapt it for other HANA installations.

Reply:
Thanks also for this information. We intentionally configure this here as this is the configuration to be used by developers in the development environment. This config file will not be included into the  project result artifact (WAR or JAR). However, all developers should have a common setup for all their machines to simplify. When the default port number is not suitable for a project, it can be easily changed after the project template has been instantiated.
@breglerj Do you have a recommended link to a guide how to `adapt it for other HANA installations`? I would add this as a comment then.

Reply:
@hohwille: Here is a link to the client documentation that describes which port to use: https://help.sap.com/viewer/0eec0d68141541d1b07893a39944924e/latest/en-US/b250e7fef8614ea0a0973d58eb73bda8.html